src/argilla/client/feedback/training/frameworks/sentence_transformers.py

```diff
@@ -0,0 +1,210 @@
+        """
+        Initializes the training arguments.
+        """
+        self.model_kwargs = {}
+
+        # Use a sample of the dataset to do some checking
+        sample = self._train_dataset[0]
+
+        if self._cross_encoder:
+            # Cross encoders don't support training with triplets
+            if (n := len(sample.texts)) > 2:
+                raise ValueError(
+                    f"Cross-encoders don't support training with triplets, the dataset has `{n}` sentences per sample."
+                )
+
+            from sentence_transformers import CrossEncoder
+
+            self._trainer_cls = CrossEncoder
+        else:
+            from sentence_transformers import SentenceTransformer
+
+            self._trainer_cls = SentenceTransformer
+
+        self.model_kwargs = get_default_args(self._trainer_cls.__init__)
+        self.trainer_kwargs = get_default_args(self._trainer_cls.fit)
+        self.dataloader_kwargs = {}
+        self.dataloader_kwargs["batch_size"] = 1
+        self._dataset_type = None
+
+        # For a guide on the selection of the loss function:
+        # https://huggingface.co/blog/how-to-train-sentence-transformers#loss-functions-for-training-a-sentence-transformers-model
+        if not self._loss_cls:
+            if sample.label:
+                if len(sample.texts) == 2:
+                    if isinstance(sample.label, int):
+                        from sentence_transformers.losses import ContrastiveLoss
+
+                        self._loss_cls = ContrastiveLoss
+                    else:
+                        from sentence_transformers.losses import CosineSimilarityLoss
+
+                        self._loss_cls = CosineSimilarityLoss
+                else:
+                    if isinstance(sample.label, int):
+                        from sentence_transformers.losses import BatchHardTripletLoss
+
+                        self._loss_cls = BatchHardTripletLoss
+                    else:
+                        from sentence_transformers.losses import BatchAllTripletLoss
+
+                        self._loss_cls = BatchAllTripletLoss
+                        from sentence_transformers.datasets import SentenceLabelDataset
+
+                        self._dataset_type = SentenceLabelDataset
+            else:
+                if len(sample.texts) == 3:
+                    from sentence_transformers.losses import TripletLoss
+
+                    self._loss_cls = TripletLoss
+                else:
+                    from sentence_transformers.losses import MultipleNegativesRankingLoss
+
+                    self._loss_cls = MultipleNegativesRankingLoss
+
+    def init_model(self) -> None:
+        """
+        Initializes a model.
+        """
+        if "model_name_or_path" in self.model_kwargs:
+            self.model_kwargs.pop("model_name_or_path")
+        if "model_name" in self.model_kwargs:
+            self.model_kwargs.pop("model_name")
```

Comment:
why are we also using `model_name` and `model_name_or_path` here?

Reply:
When we grab the default args it is done for `SentenceTransformer` and `CrossEncoder` in a generic way:
```python
self.model_kwargs = get_default_args(self._trainer_cls.__init__)
self.trainer_kwargs = get_default_args(self._trainer_cls.fit)
```
If we don't remove those argument names we have problems during the __init__ call:

```console
>       self._trainer = self._trainer_cls(self._model, **self.model_kwargs)
E       TypeError: CrossEncoder.__init__() got multiple values for argument 'model_name'
...
>       self._trainer = self._trainer_cls(self._model, **self.model_kwargs)
E       TypeError: SentenceTransformer.__init__() got multiple values for argument 'model_name_or_path'
```

In the case of `CrossEncoder` the first argument on __init__ is positional (`model_name`), while `SentenceTransformer`
uses a keyword argument (`model_name_or_path`). As we are passing this argument by position in both cases I think
it's easier to remove from the general arguments.

The same happens with the `fit` method:

```python
if "train_objectives" in self.trainer_kwargs:
    self.trainer_kwargs.pop("train_objectives")
if "train_dataloader" in self.trainer_kwargs:
    self.trainer_kwargs.pop("train_dataloader")
```

```console
E       TypeError: SentenceTransformer.fit() got multiple values for argument 'train_objectives'
...
E       TypeError: CrossEncoder.fit() got multiple values for argument 'train_dataloader'
```

Maybe there a better way of doing this that I am not seeing?

Reply:
That is the best wai to do it. I believe I misunderstood the difference of these variable per task type.